### PR TITLE
Add LogScaled metrics on Cor+

### PR DESCRIPTION
**Pull request by luisfpereira**
This PR implements the log scaled metrics on full-rank correlation matrices (see "[Permutation-invariant log-Euclidean geometries on full-rank correlation matrices](https://hal.science/hal-03878729)").

To accomplish the goal, other spaces, metrics, and diffeos are created: 

* diffeos: `LogScalingDiffeo`, `ConstantValueRowSumDiffeo`

* spaces: `NullRowSumsSymmetricMatrices`

* abstract spaces: `DiffeomorphicVectorSpace`, `DiffeomorphicMatrixVectorSpace`

* metrics: `LogScaledMetric`, `NullRowSumsPermutationInvariantMetric`

`projection` is also moved to `Manifold` to avoid inheritance problems.

Done in collaboration with @olivierbisson.

### CODE REVIEW

File: geomstats/geometry/symmetric_matrices.py [hunk 1]
```diff
@@ -341,11 +346,390 @@ def inner_product(self, tangent_vec_a, tangent_vec_b, base_point=None):
...
+
+        pre_basis = gs.array_from_sparse(indices, values, (k + 1, self.n, self.n))
+        return self.matrix_representation(
+            self.basis_representation(
+                pre_basis,
+            )
+        )
+
+
+class NullRowSumsPermutationInvariantMetric(EuclideanMetric):
```
Comment (olivierbisson) [inline geomstats/geometry/symmetric_matrices.py:622]: check the default case for n=3

**Review by olivierbisson — APPROVED** [review]
check high dimension?

File: geomstats/geometry/symmetric_matrices.py (hunk 1 above)
Comment (luisfpereira) [reply geomstats/geometry/symmetric_matrices.py:622]: Spot on! Fixed.

**Comment by luisfpereira** [comment]
> check high dimension?

For the future, this comment comes from some numerical instability related with shooting with large tangent vectors (for any metric in Cor+). We may need to check this at some point, though it seems to be stable for reasonable tangent vectors.

**Pull request merged**